proposals/0071-Entity/02-exposition-formats.md

```diff
@@ -0,0 +1,588 @@
+# Exposition Formats
+
+## Abstract
+
+This document specifies how Prometheus exposition formats should be extended to support the Entity concept introduced in [01-context.md](./01-context.md). It covers syntax additions to the text format and new protobuf message definitions.
+
+The goal is to enable first-class representation of entities—the things that produce telemetry—while maintaining backward compatibility with existing scrapers that don't understand entities.
+
+---
+
+## The Entity Concept
+
+An **Entity** represents a distinct object of interest that produces or is described by telemetry. Examples include:
+
+| Component | Description |
+|-----------|-------------|
+| **Type** | The entity type this instance belongs to (e.g., `k8s.pod`) |
+| **Identifying Labels** | Labels that uniquely identify this entity instance. Must remain constant for the entity's lifetime. |
+| **Descriptive Labels** | Additional context about the entity. May change over time. |
+
+Examples of entities:
+
+- A Kubernetes pod (`k8s.pod`) identified by namespace and UID
+- A host or node (`k8s.node`) identified by node UID
+- A service instance (`service`) identified by namespace, name, and instance ID
+
+---
+
+## Text Format
+
+### New Syntax Elements
+
+| Element | Syntax | Description |
+|---------|--------|-------------|
+| Entity type declaration | `# ENTITY_TYPE <type>` | Declares an entity type for subsequent entities |
+| Identifying labels | `# ENTITY_IDENTIFYING <label1> <label2> ...` | Lists which labels form the identity |
+| Entity instance | `<type>{<labels>}` | An entity instance (no value) |
+
+### Complete Example
+
+```
+# ENTITY_TYPE k8s.pod
+# ENTITY_IDENTIFYING k8s.namespace.name k8s.pod.uid
```

Comment:
In terms of the text format, this line seems to be the primary difference between this proposal and what an info metric currently provides.  Did you consider adding identifying labels to info metrics as an alternative?  That would address one of the current drawbacks of the new `info` function: https://prometheus.io/blog/2025/12/16/introducing-info-function/#future-development.

Reply:
Indeed. I think it is quite confusing if we create something that looks entirely new at first glance but is just a slight variation of something we had for a long time. I would rather keep the "info" nomenclature and extend the current info metric with the few missing pieces.

Also note that the `info` function could very well be the most natural way to "elevate" attributes from an entity into labels within PromQL (so that you can label-match/filter/aggregate on them). So it would be another benefit of keeping the info notion around.